CoreFindView/src/au/gov/asd/tac/constellation/views/find/advanced/QueryServices.java

```diff
@@ -0,0 +1,348 @@
+     * @see FindResult
+     * @see GraphElementType
+     */
+    public List<FindResult> quickQuery(final GraphElementType type, final String content) {
+        ReadableGraph rg = graph.getReadableGraph();
+        try {
+            final int sampleSpaceSize = type.getElementCount(rg);
+            if (sampleSpaceSize > 0) {
+                final int neededThreads = Math.min(AVAILABLE_THREADS, (int) Math.ceil((double) sampleSpaceSize / (double) MAX_THRESHOLD));
+                final int loadPerThread = (int) Math.ceil((double) sampleSpaceSize / (double) neededThreads);
+                final CyclicBarrier barrier = new CyclicBarrier(neededThreads + 1);
+
+                try {
+                    ThreadedFind[] worker = new ThreadedFind[neededThreads];
+                    // Create the requisite number of workers:
+                    for (int i = 0; i < neededThreads; i++) {
+                        final int workloadLBound = i * loadPerThread;
+                        final int workloadUBound = Math.min((sampleSpaceSize - 1), ((i + 1) * loadPerThread) - 1);
+
+                        worker[i] = new ThreadedFind(rg, barrier, this, type, content, i, workloadLBound, workloadUBound);
+
+                        // Start the worker now that it knows its workload:
+                        Thread t = new Thread(worker[i]);
+                        t.start();
+                    }
+                } finally {
+                    try {
+                        barrier.await();
+                    } catch (final InterruptedException ex) {
+                        LOGGER.log(Level.SEVERE, THREAD_INTERRUPTED, ex);
+                        Thread.currentThread().interrupt();
+                    } catch (final BrokenBarrierException ex) {
+                        LOGGER.log(Level.SEVERE, ex.getLocalizedMessage(), ex);
+                    }
+                }
+
+                return findResults; // Successfully found content.
+            }
+            return findResults;
+        } finally {
+            rg.release();
+        }
+    }
+
+    /**
+     * Selects each item in the given list on the graph.
+     *
+     * @param graph the graph on which to perform the selection.
+     * @param content A list of <code>FindResult</code>s, with each item
+     * representing one match on the graph.
+     * @param clearSelection <code>true</code> to clear previously selected
+     * items on the graph, <code>false</code> to add to them.
+     */
+    public static void selectOnGraph(final GraphWriteMethods graph, final List<FindResult> content, final boolean clearSelection) {
+        if (!clearSelection) {
+            clearSelection(graph);
+        }
+
+        final int selectedVertexAttr = graph.getAttribute(GraphElementType.VERTEX, SELECTED);
+        final int selectedLinkAttr = graph.getAttribute(GraphElementType.LINK, SELECTED);
+        final int selectedEdgeAttr = graph.getAttribute(GraphElementType.EDGE, SELECTED);
```

Comment:
curious to know whether these 2 variables ever return anything other than `Graph.NOT_FOUND`

Reply:
For simple manually generated graphs it seems to do as you suggest, only returning `Graph.NOT_FOUND`
However, there may be cases where importing or running data access plugins can generate edge/link attributes for graphs?

Reply:
From that perspective, I guess it is technically possible. That being said, it probably is worth noting that Constellation isn't particularly well set up for Edge and Link specific attributes with even the attribute editor not really looking for them.
There's probably no harm in keeping the code there for the case you've mentioned but figured I'd raise it noting that the cases where the behaviour would be different are niche at best